src/xcode/ENA/ENA/Source/Services/HealthCertificate/Models/Extensions/HealthCertificateArray+MostRelevant.swift

```diff
@@ -85,25 +85,26 @@ extension Array where Element == HealthCertificate {
 		if let currentAntigenTestCertificate = currentAntigenTestCertificate {
 			return currentAntigenTestCertificate
 		}
-
-		// Booster Vaccination Certificate
 		
-		if let boosterVaccinationCertificate = last(where: { $0.vaccinationEntry?.isBoosterVaccination ?? false }) {
-			return boosterVaccinationCertificate
-		}
-
-		// Series-completing Vaccination Certificate > 14 days
+		// Valid / Complete Vaccination Certificate
+		
+		// Booster (3/3) on Biontech, Moderna, Astra (2/2) -> gets priority
+		// Booster (2/2) on J&J (1/1) -> gets priority
 
-		let protectingVaccinationCertificate = last {
-			guard let isLastDoseInASeries = $0.vaccinationEntry?.isLastDoseInASeries, let ageInDays = $0.ageInDays else {
-				return false
-			}
-			
-			return isLastDoseInASeries && ageInDays > 14
-		}
+		// Booster with Moderna, Biontech, Astra (2/2) after Recovery Vaccination (1/1) -> gets priority after 14 days
+		// Booster with Moderna, Biontech, Astra (2/2) after J&J (1/1) -> gets priority after 14 days
 
-		if let protectingVaccinationCertificate = protectingVaccinationCertificate {
-			return protectingVaccinationCertificate
+		// Vaccination with Moderna, Biontech, Astra (1/1) after recovery -> gets priority
+		// Vaccination with J&J (1/1) after recovery -> get priority after 14 days
+		
+		if let completeVaccinationCertificate = last(where: {
+			$0.vaccinationEntry?.isBoosterVaccination ?? false || $0.vaccinationEntry?.isLastDoseInASeries ?? false && (
+			$0.ageInDays ?? 0 > 14 ||
+			$0.vaccinationEntry?.isBoosterWithBMA ?? false ||
+			$0.vaccinationEntry?.isBoosterWithJohnsonAndJohnson ?? false ||
+			$0.vaccinationEntry?.isRecoveredVaccination ?? false)
+		}) {
```

Comment:
I think this could be further simplified:
```suggestion
		if let completeVaccinationCertificate = last(where: {
			guard let vaccinationEntry = $0.vaccinationEntry else {
				return false
			}

			return vaccinationEntry.doseNumber >= vaccinationEntry.totalSeriesOfDoses && ($0.ageInDays ?? 0 > 14 || vaccinationEntry.isBoosterVaccination || vaccinationEntry.isRecoveredVaccination)
		}) {
```

That way we don't need the new `isBoosterWithBMA` and `isBoosterWithJohnsonAndJohnson` computed properties as they make no difference for the prioritisation, just using `isBoosterVaccination` is enough. Also, `isBoosterVaccination || isLastDoseInASeries` is not quite the same as the `doseNumber >= totalSeriesOfDoses` that is described in the tech spec.